File: src/Foreman.java
// Прораб: конкретний учасник, може ініціювати зміни та закривати етапи
public class Foreman extends Participant implements IProjectChangeHandler {
    private int experienceYears;

    public Foreman(int id, String name, int experienceYears) {
        super(id, name);
        this.experienceYears = experienceYears;
        System.out.println("[Foreman] Created Foreman: " + name + ", exp=" + experienceYears);
    }

    public ProjectChange initiateChange(String desc, String impact) {
        System.out.println("[Foreman] Initiating change: " + desc);
        return new ProjectChange(desc, impact);
    }

    public void closeStage() {
        System.out.println("[Foreman] Closing stage (acceptance).\n");
    }

    @Override
    public void handleChange(ProjectChange c) {
        System.out.println("[Foreman] Handling change event: " + c);
    }

    @Override
    public String getRole() {
        return "Foreman";
    }
}

File: construction/python/Foreman.py
class Foreman(Participant, IProjectChangeHandler):
    def __init__(self, id: int, name: str, experience_years: int):
        super().__init__(id, name)
        self._experience_years = experience_years
        print(f"[Foreman] Created Foreman: {name}, exp={experience_years}")

    def initiate_change(self, description: str, impact: str) -> 'ProjectChange':
        print(f"[Foreman] Initiating change: {description}")
        return ProjectChange(description, impact)

    def close_stage(self) -> None:
        print("[Foreman] Closing stage (acceptance).\n")

    def handle_change(self, change: 'ProjectChange') -> None:
        print(f"[Foreman] Handling change event: {change}")

    def get_role(self) -> str:
        return "Foreman"